File: swarmmanagercontroller/src/main/java/co/uk/swarmbit/docker/api/services/ServicesApi.java
package co.uk.swarmbit.docker.api.services;

import co.uk.swarmbit.docker.api.common.json.ServiceJson;
import co.uk.swarmbit.docker.api.services.parameters.*;
import co.uk.swarmbit.docker.api.common.annotation.DockerRemoteApiMinVersion;
import co.uk.swarmbit.docker.api.common.json.ServiceGeneralResponseJson;

import java.util.List;

public interface ServicesApi {

    List<ServiceJson> listServices(String swarmId, ServicesListParameters parameters);

    ServiceJson inspectService(String swarmId, String id, ServiceInspectParameters parameters);

    ServiceGeneralResponseJson createService(String swarmId, ServiceCreateParameters parameters);

    void updateService(String swarmId, String id, ServiceUpdateParameters parameters);

    void deleteService(String swarmId, String id);

    @DockerRemoteApiMinVersion("v1.29")
    byte[] getServiceLogs(String swarmId, String id, ServiceLogsParameters parameters);

}

File: swarmmanagercontroller-new/src/main/kotlin/com/swarmbit/docker/api/services/ServicesApi.kt
package com.swarmbit.docker.api.services

import com.swarmbit.docker.api.common.annotation.DockerRemoteApiMinVersion
import com.swarmbit.docker.api.services.model.Service
import com.swarmbit.docker.api.services.model.ServiceGeneralResponse
import com.swarmbit.docker.api.services.parameters.ServicesCreateParameters
import com.swarmbit.docker.api.services.parameters.ServicesInspectParameters
import com.swarmbit.docker.api.services.parameters.ServicesListParameters
import com.swarmbit.docker.api.services.parameters.ServicesLogsParameters
import com.swarmbit.docker.api.services.parameters.ServicesUpdateParameters

interface ServicesApi {

    fun listServices(swarmId: String, parameters: ServicesListParameters): List<Service>

    fun inspectService(swarmId: String, id: String, parameters: ServicesInspectParameters): Service

    fun createService(swarmId: String, parameters: ServicesCreateParameters): ServiceGeneralResponse

    fun updateService(swarmId: String, id: String, parameters: ServicesUpdateParameters)

    fun deleteService(swarmId: String, id: String)

    @DockerRemoteApiMinVersion("v1.29")
    fun getServiceLogs(swarmId: String, id: String, parameters: ServicesLogsParameters): ByteArray
}
